[acsone-addons][event_mass_mailing][NEW]

JonathanNEMRY opened the pull request:
Introduce and Use the mass mailing concept to event management

Commit:
[acsone-addons][event_mass_mailing][NEW]
Introduce and Use the mass mailing concept to event management

lmignon commented on event_mass_mailing/models/event.py:
```diff
@@ -0,0 +1,148 @@
+
+        return create, mailing
+
+    _inherit = "event.event"
```
_inherit should be just after the class declaration

lmignon commented on event_mass_mailing/models/event.py:
```diff
@@ -0,0 +1,148 @@
+        'email.template', string='Cancellation Template',
+        select=True, track_visibility='onchange')
+    cancellation_date = fields.Datetime(string="Cancellation Date")
+
```
IMO, code readability will be improved if you move the declarations of the fields at the beginning of the class, just afet _inherit

JonathanNEMRY replied:
I have applied those changes and will conserve this norms for future
Thanks for your review!

lmignon replied:
It's seem that in odoo, by convention(for python code), docstrings are quoted using double quotes (") and others string with single quote (').  Whatever your choice, try to stay consistent. ex: 

``` python
confirm_mass_mailing_id = fields.Many2one(
        'mail.mass_mailing', string='Mass Mailing Confirmation',
        select=True, track_visibility='onchange')
confirmation_date = fields.Datetime(string="Confirmation Date")
```

``` python
    _inherit = "event.registration"
    _mail_mass_mailing = 'Event Registration'
```

I know, it's a details and I am not myself always consistent. .. :smile:

Commit:
[event_mass_mailing][CHG] Review code design and add some comments

Commit:
[acsone-addons][event_mass_mailing][CHG] Improve code consistency by using ' for string and " for docstrings

lmignon commented:
:+1:  (code review only)

Commit:
[event_mass_mailing][FIX] auto-install False

Olivier-LAURENT commented on event_mass_mailing/models/event.py:
```diff
@@ -0,0 +1,162 @@
+    invitation_template_id = fields.Many2one(
+        'email.template', string='Invitation Template',
+        select=True, track_visibility='onchange')
+    invite_mass_mailing_id = fields.Many2one(
```
Maybe it is better to track invitation_date instead of this generated readonly field...

JonathanNEMRY replied:
It's a mistake, those *_date should be track. I'll Fix this

Olivier-LAURENT commented on event_mass_mailing/models/event.py:
```diff
@@ -0,0 +1,162 @@
+    # confirmation
+    confirm_mass_mailing_id = fields.Many2one(
+        'mail.mass_mailing', string='Mass Mailing Confirmation',
+        select=True, track_visibility='onchange')
```
idem

Olivier-LAURENT commented on event_mass_mailing/models/event.py:
```diff
@@ -0,0 +1,162 @@
+    # cancellation
+    cancel_mass_mailing_id = fields.Many2one(
+        'mail.mass_mailing', string='Mass Mailing Cancellation',
+        select=True, track_visibility='onchange')
```
idem

Olivier-LAURENT commented on event_mass_mailing/models/event.py:
```diff
@@ -0,0 +1,162 @@
+            self.invitation_template_id, self.invite_mass_mailing_id, domain)
+        if create:
+            self.invite_mass_mailing_id = mailing
+        self.invitation_date = date.today().strftime(format)
```
cf. fields.Date.today()

JonathanNEMRY replied:
Indeed it seems better with `fields` option.
However I'll fix it with `fields.Dateime.now()` cause I use a `fields.Datetime`
tks

Olivier-LAURENT commented on event_mass_mailing/models/event.py:
```diff
@@ -0,0 +1,162 @@
+            self.email_confirmation_id, self.confirm_mass_mailing_id, domain)
+        if create:
+            self.confirm_mass_mailing_id = mailing
+        self.confirmation_date = date.today().strftime(format)
```
idem

Olivier-LAURENT commented on event_mass_mailing/models/event.py:
```diff
@@ -0,0 +1,162 @@
+            self.cancellation_template_id, self.cancel_mass_mailing_id)
+        if create:
+            self.cancel_mass_mailing_id = mailing
+        self.cancellation_date = date.today().strftime(format)
```
idem

Olivier-LAURENT commented on event_mass_mailing/models/event.py:
```diff
@@ -0,0 +1,162 @@
+        regs = self.registration_ids.filtered(
+            lambda reg: reg.state in states)
+        if not regs:
+            raise Warning(_('Send Invitation Failed: no Registrations '
```
warning only valid for invitations but the method is also called in the context of confirmation and cancelation

Olivier-LAURENT commented on event_mass_mailing/models/event.py:
```diff
@@ -0,0 +1,162 @@
+        """
+        self._check_registrations(['open'])
+        domain = ('state', 'not in', ['draft', 'cancel'])
+        create, mailing = self._mass_mailing_action(
```
without calling the super() method it is necessary to update the event state

Commit:
[event_mass_mailing]
[CHG]
- Date format is now fields.Datetime format
- Track *_date fields
[FIX]
- update state when confirm/cancell due to a redefinition of native method
- Add fix case into tests

Commit:
[event_mass_mailing][CHG] do not track mailing

Commit:
[event_mass_mailing][FIX] super set state to cancel for event and for registrations

Olivier-LAURENT commented:
:+1:
